app/career/models/weekly_business.py

```diff
@@ -0,0 +1,39 @@
+import uuid
+
+from django.db import models
+
+from app.util import today
+from app.util.models import BaseModel, OptionalImage
+
+
+class WeeklyBusiness(BaseModel, OptionalImage):
+    id = models.UUIDField(
+        auto_created=True, primary_key=True, default=uuid.uuid4, serialize=False,
+    )
+
+    business_name = models.CharField(max_length=200)
+    body = models.TextField()
+
+    year = models.PositiveSmallIntegerField()
+    week = models.PositiveSmallIntegerField()
+
+    class Meta:
+        unique_together = ("year", "week")
+        verbose_name = "Weekly business"
+        verbose_name_plural = "Weekly businesses"
+        ordering = ["-year", "-week"]
+
+    def __str__(self):
+        return f"{self.business_name} - {self.year} - {self.week}"
+
+    def save(self, *args, **kwargs):
+        if self.week < 1 or self.week > 52:
+            raise ValueError("Uke må være mellom 1 og 52")
+        if self.year < today().year:
+            raise ValueError("Ukens bedrift kan ikke opprettes i fortiden")
+        try:
+            if WeeklyBusiness.objects.get(week=self.week, year=self.year).id != self.id:
```

Comment:
Is it necessary to have this logic here? Django will raise an exception anyways which might be clearer to use at a higher level. I feel like this logic is a bit out of scope for a save-method on a model. 

It might be more clear to create an exception class `WeeklyBusinessAlreadyExistsException` as well.

What do you think?

Reply:
this is for the error message only. Makes it easier for giving message for frontend. And since this is a valueerror i dont really see the need for a separate exception

Reply:
I don't disagree 😉